README.md

```diff
@@ -2,31 +2,308 @@
+turn, tells Java that we are to use the double quote within the `String` and not
+end the `String` (since we usually wrap `String` values in double quotes). Let's
+look at an example of this:
+
+```java
+System.out.println("\"You must be the change you wish to see in the world.\" - Mahatma Gandhi");
+```
+
+Output:
+
+```text
+"You must be the change you wish to see in the world." - Mahatma Gandhi
+```
+
+Notice how the backslashes are not printed but the double quote immediately
+following the backslash is printed. This is because the escape character
+notifies Java that we are to print the escape sequence instead that results in
+just a double quotation mark. Java also treats the escape sequence as one
+character - meaning `\"` would be considered one character, not two.
+
+Consider the table of escape sequences:
+
+| Escape Sequence | Display      |
+|-----------------|--------------|
+| `\t`            | Tab          |
+| `\n`            | New Line     |
+| `\'`            | Single Quote |
+| `\"`            | Double Quote |
+| `\\`            | Backslash    |
+
+A few notes on the escape sequences: The tab and new line characters are
+representatives of whitespace. Consider the example below of how to use these
+characters:
+
+```java
+String firstString = "Hello\tWorld!";
+String secondString = "What a beautiful\nday!";
+
+System.out.println(firstString);
+System.out.println(secondString);
+```
+
+Output:
+
+```plaintext
+Hello   World!
+What a beautiful
+day!
+```
+
+We can see in the above output that when we placed the character `\t` between
+"Hello" and "World!", it placed a tab or an indent between the two words. In the
+`secondString`, we place a `\n` character between the words "beautiful" and
+"day!". When we print that to the screen, it will actually write the word "day!"
+on a new line separate from "What a beautiful".
+
+## Strings and Character Arrays
+
+A variable declared with data type `String` stores a sequence of characters.
+
+A `String literal` is a sequence of characters surrounded with double quotes,
+such as "Hello" or "123 Main St".
+
+A sequence of characters can be treated as a single unit... and that is
+where the `String` data type comes in. But we also learned about arrays. So
+wouldn't an array of characters be the same thing as a `String`?
+
+The answer is: no - but pretty close!
+
+In Java, there are a few differences between a `String` and a `char[]`:
+
+1. A `String` is a sequence of characters where a `char[]` is a collection of
+   `char` data types. In a `char[]`, each element could be treated independently
+   and separately. For example, we could have a character array of grades:  
+   `char[] grades = {'A', 'B', 'C', 'D', 'F'};`    
+   Since it is an array, we can
+   change the element at a particular position in the array. For example, we can
+   reassign the first element as `grades[0] = 'Z'`;
+2. A `String` is an immutable type whereas a `char[]` is still mutable.
```

Comment:
I added a brief definition of immutable.  Let me know if more should be added.

Reply:
I think this is fine for now. Maybe when we get to the lesson about constructing strings we provide them the definition of immutable again and really showcase it there.

Reply:
ok.  I also mention it a lot in the string manipulation lesson by showing memory always creating a new string for each method, rather than changing the existing string.